pike/path.py

```diff
@@ -0,0 +1,690 @@
+
+        :type file_information_class: smb2.FileInformationClass
+        :param file_information_class: the file information class constant
+            describing the type of information to retrieve
+        :type info_type: smb2.InfoType
+        :param info_type: the SMB2 query info type
+        :type options: smb2.CreateOptions
+        :param options: options passed to CREATE when opening the file
+        :return: single file info object corresponding to the
+            file_information_class requested
+        """
+        options = options | smb2.FILE_OPEN_REPARSE_POINT
+        return self.stat(
+            file_information_class=file_information_class,
+            info_type=info_type,
+            options=options,
+        )
+
+    def chmod(self, mode):
+        raise NotImplementedError("ACL modification is not supported")
+
+    lchmod = chmod
+
+    def exists(self, options=0):
+        """
+        :type options: smb2.CreateOptions
+        :param options: options passed to CREATE when opening the file
+        :return: True if this path exists
+        """
+        try:
+            with self._create(
+                access=0,
+                disposition=smb2.FILE_OPEN,
+                options=options,
+            ) as handle:
+                return True
+        except model.ResponseError as re:
+            if re.response.status not in NOT_FOUND_STATUSES + (
+                ntstatus.STATUS_FILE_IS_A_DIRECTORY,
+                ntstatus.STATUS_NOT_A_DIRECTORY,
+            ):
+                raise
+        return False
+
+    def expanduser(self):
+        raise NotImplementedError("expanduser() is not supported")
+
+    def group(self):
+        raise NotImplementedError("group() is not supported")
+
+    def is_dir(self):
+        """
+        :return: True if this path is a directory
+        """
+        return self.exists(options=smb2.FILE_DIRECTORY_FILE)
+
+    def is_file(self):
+        """
+        :return: True if this path is a normal file
+        """
+        return self.exists(options=smb2.FILE_NON_DIRECTORY_FILE)
+
+    def is_symlink(self):
+        """
+        :return: True if this path is a reparse point
+        """
+        try:
+            with self._create(
+                access=0,
+                disposition=smb2.FILE_OPEN,
+                _follow=False,
+            ) as handle:
+                pass
+        except model.ResponseError as re:
+            if re.response.status == ntstatus.STATUS_STOPPED_ON_SYMLINK:
+                return True
+            if re.response.status not in NOT_FOUND_STATUSES:
+                raise
+        return False
+
+    def is_mount(self):
+        """
+        Note: all PikePath instances are considered to be mounts
+
+        :return: True if this path is a mount point
+        """
+        return True  # all paths are on SMB mount
+
+    def is_socket(self):
+        """
+        Note: no PikePath instances are considered to be sockets
```

Comment:
What about `IPC$`?  Closer to being a socket than anything else.  Or perhaps there is already precedent for determing `IPC$` type?

Reply:
a socket in this case refers to a unix socket; but point taken about ipc$:

I update the `is_fifo` method to check whether the Tree path `.endwith("\\ipc$")`.

Pike doesn't currently do anything with named pipes today, but it could in the future.